### PR TITLE
feat: startup new llamalend table

### PR DESCRIPTION
**Pull request by DanielSchiavini**
- implement new llamalend table (hidden in production)
- Table is visible even without connecting wallet ([preview](https://curve-dapp-crvusd-git-feat-llamalend-curvefi.vercel.app/#/ethereum/beta-markets))
- ~~depends on #492~~

### CODE REVIEW

File: apps/loan/src/entities/vaults.ts
Comment (0xAlunara) [inline apps/loan/src/entities/vaults.ts:4]: They're essentially the same, but any reason why you went with interfaces instead of types like you did in `crvusd.ts` and `lending.ts` in the same folder?

File: apps/loan/src/entities/lending.ts
Comment (0xAlunara) [inline apps/loan/src/entities/lending.ts:55]: A whole day damn

File: apps/loan/src/components/PageLlamaMarkets/LendTableFooter.tsx
Comment (0xAlunara) [inline apps/loan/src/components/PageLlamaMarkets/LendTableFooter.tsx:25]: Have you considered making the header and button part of the grid with a size 12? Don't forget to set the grid's `rowGap` to `Spacing.sm`.

That way I think you can get rid of the `<Box>` and `width: 100%` in the button below.

File: apps/loan/src/components/PageLlamaMarkets/LendTableFooter.tsx
Comment (0xAlunara) [inline apps/loan/src/components/PageLlamaMarkets/LendTableFooter.tsx:26]:
```ts
paddingInline: Spacing.lg
paddingBlockStart: Spacing.lg
paddingBlockEnd: Spacing.sm // Not sure why but it's what the Figma says
```

File: apps/loan/src/components/PageLlamaMarkets/LendTableFooter.tsx
Comment (0xAlunara) [inline apps/loan/src/components/PageLlamaMarkets/LendTableFooter.tsx:19]: Needs `IconSize.xxl`

File: apps/loan/src/components/PageLlamaMarkets/LendTableFooter.tsx
Comment (0xAlunara) [inline apps/loan/src/components/PageLlamaMarkets/LendTableFooter.tsx:45]: Use the new "link" variant, also gets rid of that ugly focus outline once you click.

File: apps/loan/src/components/PageLlamaMarkets/cells/SupplyYieldCell.tsx
Comment (0xAlunara) [inline apps/loan/src/components/PageLlamaMarkets/cells/SupplyYieldCell.tsx:3]: All those number cells with percentages and dollars should probably use the `<Metric>` component in the future (it's a WIP) with all the bells and whistles turned off. Just so that there's consistency in unit font rendering and fractional rendering. But whatever, WIP, this is good enough for now.

File: apps/loan/src/components/PageLlamaMarkets/LendTableFooter.tsx
Comment (0xAlunara) [inline apps/loan/src/components/PageLlamaMarkets/LendTableFooter.tsx:28]: As suggested above you might get rid of this Box, but if you don't, not sure where the `Spacing.md` is in the Figma design, I can't find it anywhere.

File: apps/loan/src/components/PageLlamaMarkets/LendTableTitle.tsx
Comment (0xAlunara) [inline apps/loan/src/components/PageLlamaMarkets/LendTableTitle.tsx:5]: Why not use `import { styled } from '@mui/material/styles'`, then you can create a styled  with sx properties.

File: apps/loan/src/components/PageLlamaMarkets/LendTableTitle.tsx
Comment (0xAlunara) [inline apps/loan/src/components/PageLlamaMarkets/LendTableTitle.tsx:21]: You already commented this in the Figma, but these diff card shadows with hardcoded colors are are nasty to have.

File: apps/loan/src/components/PageLlamaMarkets/LendTableTitle.tsx
Comment (0xAlunara) [inline apps/loan/src/components/PageLlamaMarkets/LendTableTitle.tsx:28]: I don't see no gap in the Figma

File: apps/loan/src/components/PageLlamaMarkets/LendTableTitle.tsx
Comment (0xAlunara) [inline apps/loan/src/components/PageLlamaMarkets/LendTableTitle.tsx:30]: I don't think you need this

File: apps/loan/src/components/PageLlamaMarkets/LendTableTitle.tsx
Comment (0xAlunara) [inline apps/loan/src/components/PageLlamaMarkets/LendTableTitle.tsx:18]: Personally I don't see the benefit in destructuring this, now there's a combination of `md` and `Spacing.xx` where xx something you've not destructured.

File: apps/loan/src/components/PageLlamaMarkets/LendTableTitle.tsx
Comment (0xAlunara) [inline apps/loan/src/components/PageLlamaMarkets/LendTableTitle.tsx:39]: flex="1 1 50%" can be removed I think (not 100% sure)

File: packages/curve-ui-kit/src/shared/ui/TableFilters.tsx
Comment (0xAlunara) [inline packages/curve-ui-kit/src/shared/ui/TableFilters.tsx:20]: `{SizesAndSpaces.Grid.Column_Spacing}`

**Review by 0xAlunara — CHANGES_REQUESTED** [review]

File: packages/curve-ui-kit/src/themes/design/1_sizes_spaces.ts
Comment (DanielSchiavini) [reply packages/curve-ui-kit/src/themes/design/1_sizes_spaces.ts:275]: These are supposed to be responsive. I created a task for fixing it

File: packages/curve-ui-kit/src/shared/routes.ts
Comment (DanielSchiavini) [reply packages/curve-ui-kit/src/shared/routes.ts:35]: I think we can have a single isBeta for all features

File: packages/curve-ui-kit/src/shared/routes.ts
Comment (DanielSchiavini) [reply packages/curve-ui-kit/src/shared/routes.ts:37]: This is just for production 
Other hosts always show beta features rn

File: apps/loan/src/entities/vaults.ts
Comment (DanielSchiavini) [reply apps/loan/src/entities/vaults.ts:91]: We can remove it

File: apps/loan/src/entities/lending.ts
Comment (DanielSchiavini) [reply apps/loan/src/entities/lending.ts:55]: The data contains one entry per day at midnight

File: apps/loan/src/entities/vaults.ts
Comment (DanielSchiavini) [reply apps/loan/src/entities/vaults.ts:4]: Types are fine. These were generated

File: packages/curve-ui-kit/src/shared/ui/TableGrid.tsx
Comment (DanielSchiavini) [reply packages/curve-ui-kit/src/shared/ui/TableGrid.tsx:1]: We can consider something. Do you know any nice ones?

Comment (0xAlunara) [reply packages/curve-ui-kit/src/shared/ui/TableGrid.tsx:1]: I don't have any personal experience with them as I'm more familiar with Vue and my own implementation, but I think for Curve there's 3 major products to consider:

1. [TanStack Table](https://tanstack.com/table/latest)
2. [RadixUI](https://www.radix-ui.com/themes/docs/components/table)
3. [Shadcn](https://ui.shadcn.com/docs/components/data-table)

Given we already use TanStack Query I think we should take a look at TanStack Table. At first glance it looks well engineered, feature rich, supports React very well and is also headless so we can style it.

The alternatives are RadixUI and Shadcn. The reason these might be interesting is that you don't need a whole new framework, but can just copy paste the Data table component. RadixUI's version is the vanilla one, but I don't know if it contains all features we'd need. Shadcn built's on top of that version and adds features and such, but Shadcn's schtick is it's also built around Tailwind, which makes it less suitable for us.

Comment (DanielSchiavini) [reply packages/curve-ui-kit/src/shared/ui/TableGrid.tsx:1]: Ideally we should reuse the mui work and not have to customize a whole  new framework :disappointed:

File: packages/curve-ui-kit/src/themes/design/1_sizes_spaces.ts
Comment (DanielSchiavini) [reply packages/curve-ui-kit/src/themes/design/1_sizes_spaces.ts:311]: this was the only merge conflict with master so easy fix

File: packages/curve-ui-kit/src/themes/design/1_sizes_spaces.ts
Comment (DanielSchiavini) [reply packages/curve-ui-kit/src/themes/design/1_sizes_spaces.ts:263]: I used the desktop values now, they should be responsive: https://www.notion.so/Make-typography-responsive-15b599aae06480d5a4e5e70c7bf5ccc0?pvs=4

File: packages/curve-ui-kit/src/shared/routes.ts
Comment (DanielSchiavini) [reply packages/curve-ui-kit/src/shared/routes.ts:35]: I don't see the need to enable/disable flags for each feature

File: apps/loan/src/components/PageLlamaMarkets/LendTableFooter.tsx
Comment (DanielSchiavini) [reply apps/loan/src/components/PageLlamaMarkets/LendTableFooter.tsx:25]: It's a good idea, it makes the component simpler, but I couldn't get rid of the 100% width

File: apps/loan/src/components/PageLlamaMarkets/LendTableFooter.tsx
Comment (DanielSchiavini) [reply apps/loan/src/components/PageLlamaMarkets/LendTableFooter.tsx:45]: This one needs 48px (large size) the link is 32px only

File: apps/loan/src/components/PageLlamaMarkets/cells/SupplyYieldCell.tsx
Comment (DanielSchiavini) [reply apps/loan/src/components/PageLlamaMarkets/cells/SupplyYieldCell.tsx:3]: Good point, I'll use it once merged

File: apps/loan/src/components/PageLlamaMarkets/LendTableFooter.tsx
Comment (DanielSchiavini) [reply apps/loan/src/components/PageLlamaMarkets/LendTableFooter.tsx:28]: It's here. I replaced with rowGap as you suggested
image

File: apps/loan/src/components/PageLlamaMarkets/LendTableTitle.tsx
Comment (DanielSchiavini) [reply apps/loan/src/components/PageLlamaMarkets/LendTableTitle.tsx:5]: aha I didn't know why I missed the sx :laughing:

File: apps/loan/src/components/PageLlamaMarkets/LendTableTitle.tsx
Comment (DanielSchiavini) [reply apps/loan/src/components/PageLlamaMarkets/LendTableTitle.tsx:28]: yah there was a problem with the lineHeight being incorrect, fixed now

File: apps/loan/src/components/PageLlamaMarkets/LendTableTitle.tsx
Comment (DanielSchiavini) [reply apps/loan/src/components/PageLlamaMarkets/LendTableTitle.tsx:18]: yeah it was about the comment above. I'll just remove this for now

File: apps/loan/src/components/PageLlamaMarkets/LendTableTitle.tsx
Comment (DanielSchiavini) [reply apps/loan/src/components/PageLlamaMarkets/LendTableTitle.tsx:30]: replaced with flexGrow on the 2nd one

File: packages/curve-ui-kit/src/shared/ui/TableGrid.tsx
Comment (DanielSchiavini) [reply packages/curve-ui-kit/src/shared/ui/TableGrid.tsx:1]: We'll discuss this together and make a separate PR

**Review by 0xAlunara — CHANGES_REQUESTED** [review]

File: packages/curve-ui-kit/src/widgets/Footer/Footer.tsx
Comment (DanielSchiavini) [reply packages/curve-ui-kit/src/widgets/Footer/Footer.tsx:39]: Oh damn , sorry for that.

File: packages/curve-common/src/widgets/Header/MobileHeader.tsx
Comment (DanielSchiavini) [inline packages/curve-common/src/widgets/Header/MobileHeader.tsx:55]: This was actually a bug, because it would go behind the scrollbar, causing a horizontal scroll. Caught by the new cypress tests.

**Review by 0xAlunara — APPROVED** [review]

**Review by 0xtutti — APPROVED** [review]

**Review by 0xtutti — APPROVED** [review]

**Review by 0xAlunara — APPROVED** [review]

**Review by 0xtutti — APPROVED** [review]

**Pull request merged**
